File: 289.java
public class Solution {
    public void gameOfLife(int[][] board) {
        int m = board.length;
        int n = board[0].length;
        int[][] directions = {{-1, -1}, {-1, 0}, {-1, 1}, {0, -1}, {0, 1}, {1, -1}, {1, 0}, {1, 1}};
        int[][] tempBoard = new int[m][n];
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                tempBoard[i][j] = board[i][j];
            }
        }
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                int liveNeighbors = 0;
                for (int[] dir : directions) {
                    int row = i + dir[0];
                    int col = j + dir[1];
                    if (row >= 0 && row < m && col >= 0 && col < n && tempBoard[row][col] == 1) {
                        liveNeighbors++;
                    }
                }
                if (tempBoard[i][j] == 1) {
                    if (liveNeighbors < 2 || liveNeighbors > 3) {
                        board[i][j] = 0;
                    }
                } else {
                    if (liveNeighbors == 3) {
                        board[i][j] = 1;
                    }
                }
            }
        }
    }
}

File: 289.py
from typing import List

class Solution:
    def gameOfLife(self, board: List[List[int]]) -> None:
        m = len(board)
        n = len(board[0])
        directions = [[-1, -1], [-1, 0], [-1, 1], [0, -1], [0, 1], [1, -1], [1, 0], [1, 1]]
        tempBoard = [[board[i][j] for j in range(n)] for i in range(m)]
        
        for i in range(m):
            for j in range(n):
                live_neighbors = 0
                # Count the live neighbors
                for entry in directions:
                    row = i + entry[0]
                    col = j + entry[1]
                    if (row >= 0 and row < m and col >= 0 and col < n and tempBoard[row][col] == 1):
                        live_neighbors += 1

                if tempBoard[i][j] == 1:
                    if live_neighbors < 2 or live_neighbors > 3:
                        board[i][j] = 0
                else:
                    if live_neighbors == 3:
                        board[i][j] = 1 

s = Solution()
board = [[0, 1, 0], [0, 0, 1], [1, 1, 1], [0, 0, 0]]
s.gameOfLife(board)
print(board)